### ISSUE TITLE
Thanks for this

### ISSUE THREAD

**Issue by tyeeman** [Sorrow446/ZS-DL#1]
Yes, much appreciated, but will this need updating over and over when Zippy changes, like the repo that motivated you to do this?

**Comment by Sorrow446** [comment]
Welcome. Yup. I plan to add a JS alternative some time in the future. Speed will be sacrificed of course. I'm not sure how frequently ZS change their method, but I'll keep this updated.

**Comment by tyeeman** [comment]
Great, thanks. FMI, since I'm not a coder or regex expert are those two things required to update your code? I see around line 84 the regex comes into play. Yeah, I guess you have to be an expert to update it.

**Comment by tyeeman** [comment]
I'm only getting one of 6 urls working, the last one here, please check - 
```
https://www67.zippyshare.com/v/ZRvsTA0p/file.html
https://www67.zippyshare.com/v/psvXkngU/file.html
https://www67.zippyshare.com/v/9UzuuoX2/file.html
https://www67.zippyshare.com/v/XM1EEQ2D/file.html
https://www67.zippyshare.com/v/606u3EBb/file.html
https://www67.zippyshare.com/v/QqzBqooz/file.html
```

**Comment by Sorrow446** [comment]
I was thinking about that. I'll add a switch to dump them instead of download. Maybe using the -o switch actually. Something like, `-o out.txt`

**Comment by tyeeman** [comment]
Did you confirm only the last link above works?

**Comment by Sorrow446** [comment]
All are working here. What was the issue?

**Comment by tyeeman** [comment]
<img width="500" alt="2020-05-30_105117" src="https://user-images.githubusercontent.com/4469306/83335733-9c8a6900-a263-11ea-8fac-3b5f77330136.png">

**Comment by Sorrow446** [comment]
Could you do a search in your browser on the HTML for this? 

`document.getElementById('dlbutton').href = "/d/ZRvsTA0p/" + (541881 % 51245 + 541881 % 913) + "/Rian.Ruined.Star.Wars.part1.rar";`

**Comment by tyeeman** [comment]
Forgive me, but your app doesn't use my browser so I'm a little confused. You want me to type in the address bar ?

**Comment by Sorrow446** [comment]
Go to the first URL in your browser, then do a search with the F12 key.

**Comment by tyeeman** [comment]
OK here it is - 
```
document.getElementById('dlbutton').href = "/d/ZRvsTA0p/" + (764230 % 51245 + 764230 % 913) + "/Rian.Ruined.Star.Wars.part1.rar";
if (document.getElementById('fimage')) {
    document.getElementById('fimage').href = "/i/ZRvsTA0p/" + (764230 % 51245 + 764230 % 913) + "/Rian.Ruined.Star.Wars.part1.rar";
}

```

**Comment by Sorrow446** [comment]
It's still a full regex match. Not sure what's going on here.

**Comment by tyeeman** [comment]
I downloaded the latest and it matches my current version exactly.

**Comment by Sorrow446** [comment]
Are you passing each URL with a single space inbetween?

**Comment by tyeeman** [comment]
I'm using a txt file with one url per line. I guess that's wrong then. I don't see a txt file example in the readme?

**Comment by Sorrow446** [comment]
One per line is fine. There is. 

> Download from text file to "G:\ZS-DL downloads" with HTTPS proxy:
> ZS-DL.py -u G:\links.txt -o "G:\ZS-DL downloads" -p 0.0.0.0:8080

**Comment by tyeeman** [comment]
I get this error - 
```
IndentationError: unindent does not match any outer indentation level
```

**Comment by tyeeman** [comment]
OK, tabs restored and it runs. I was going to say everything scrolls off my cmd window, which most does  and I couldn't find getElementById. Just for fun I deleted all links except the first one and it did start to download (part1), so it looks like only the last file in my links.txt file works. I left 2 links in the links.txt file and only the second one started to download. Does this help?

**Comment by tyeeman** [comment]
Ok, we're getting closer, it seems to be related to the end of line character. I just added a line feed character after the last link in links.txt and nothing would download so definitely related.

**Comment by Sorrow446** [comment]
Fixed :). https://github.com/Sorrow446/ZS-DL/commit/7574da8dfff73c477ec598ed2cb52d7ec9cf12c8

**Comment by tyeeman** [comment]
Working great now, thanks!

**Comment by PaulinoRBJ** [comment]
@Sorrow446 Thanks for the script! Do you plan on adding support for dlc containers?

**Comment by Sorrow446** [comment]
@PaulinoRBJ https://github.com/Sorrow446/ZS-DL/commit/a08e57fff4faf5bc36e51d2f9ab9964e85dd3119

Please try it out.

**Comment by PaulinoRBJ** [comment]
@Sorrow446 Thanks! It works great. However there seems to be an issue if more than 2 DLCs are specified- `zs-dl.py` seems to handle only the first 2 two as DLC containers while the others are treated as URL's.  Here's what it looks like:
```
> zs-dl.py -u *.dlc

 _____ _____     ____  __
|__   |   __|___|    \|  |
|   __|__   |___|  |  |  |__
|_____|_____|   |____/|_____|

Processing DLC container: container1.dlc
Processing DLC container: container2.dlc

URL 1 of 127:
URL failed.
ValueError: Invalid URL: container3.dlc

URL 2 of 127:
URL failed.
ValueError: Invalid URL: container4.dlc

URL 3 of 127:
container1.part01.rar
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 500M/500M [01:21<00:00, 6.42MB/s]
.
.
```

**Comment by Sorrow446** [comment]
@PaulinoRBJ Do you have the latest ver? Forgot a period. https://github.com/Sorrow446/ZS-DL/commit/f41dc08455b3957d09b083910d1092eba6575c2a

**Comment by PaulinoRBJ** [comment]
@Sorrow446 Yes, I am using the latest post  [`f41dc08`](https://github.com/Sorrow446/ZS-DL/commit/f41dc08455b3957d09b083910d1092eba6575c2a) version.

**Comment by Sorrow446** [comment]
@PaulinoRBJ Damn. Could you send those links so I could reproduce?

**Comment by PaulinoRBJ** [comment]
@Sorrow446 sure. I've attached 3 test DLCs I made and then tested with the same results I was getting before. 
[test-DLCs.zip](https://github.com/Sorrow446/ZS-DL/files/4726518/test-DLCs.zip)
```
➜  ~ zs-dl.py -u *.dlc

 _____ _____     ____  __
|__   |   __|___|    \|  |
|   __|__   |___|  |  |  |__
|_____|_____|   |____/|_____|

Processing DLC container: testf1.dlc
Processing DLC container: testf3.dlc

URL 1 of 3:
URL failed.
ValueError: Invalid URL: testf2.dlc

URL 2 of 3:
testf1
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3.00/3.00 [00:00<00:00, 5.07kB/s]

URL 3 of 3:
testf3
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3.00/3.00 [00:00<00:00, 4.01kB/s]
```

**Comment by Sorrow446** [comment]
Fixed. Sorry for the trouble.

**Comment by PaulinoRBJ** [comment]
@Sorrow446 thanks again for this fix, DLCs are now working perfectly. Wasn't any trouble and I love that I was able to help in some way. 😄

**Comment by abdulaziz-git** [comment]
@Sorrow446 the pattern is change again, please fix
Thanks

**Comment by Sorrow446** [comment]
Fixed. Pushed to binary and source.

**Comment by rcdosado** [comment]
zippy updates the pattern again :(

**Comment by abdulaziz-git** [comment]
> zippy updates the pattern again :(

Yes change the pattern again
@Sorrow446 can you show me how you check new pattern if zippyshare change the pattern?

**Comment by Sorrow446** [comment]
> @Sorrow446 the pattern is change again, please fix
> Thanks

Fixed. Binary & source.
> > zippy updates the pattern again :(
> 
> Yes change the pattern again
> @Sorrow446 can you show me how you check new pattern if zippyshare change the pattern?

Right click -> view page source. Do a search for "document.getElementById('dlbutton')".
